jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/server/ServerSpanDecorator.java

```diff
@@ -0,0 +1,103 @@
+package io.opentracing.contrib.jaxrs.server;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.core.MultivaluedMap;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.jaxrs.internal.URIUtils;
+import io.opentracing.tag.Tags;
+
+/**
+ * @author Pavol Loffay
+ */
+public interface ServerSpanDecorator {
+
+    /**
+     * Decorate span by incoming object.
+     *
+     * @param requestContext
+     * @param span
+     */
+    void decorateRequest(ContainerRequestContext requestContext, Span span);
+
+    /**
+     * Decorate spans by outgoing object.
+     *
+     * @param responseContext
+     * @param span
+     */
+    void decorateResponse(ContainerResponseContext responseContext, Span span);
+
+    /**
+     * Adds standard tags: {@link io.opentracing.tag.Tags#SPAN_KIND},
+     * {@link io.opentracing.tag.Tags#HTTP_METHOD}, {@link io.opentracing.tag.Tags#HTTP_URL} and
+     * {@link io.opentracing.tag.Tags#HTTP_STATUS}
+     */
+    ServerSpanDecorator STANDARD_TAGS = new ServerSpanDecorator() {
```

Comment:
I would prefer to make the SPAN_KIND tag both (1) built-in to this package (i.e., non-negotiable) and (2) provided at `tracer.buildSpan(...)`-time.

Part of the rationale is that we *know* this tag should be applied... the other piece is that some impls actually change behavior around SpanContext ids depending on the presence of the SERVER tag *at startSpan() time*. E.g., some zipkin tracers want to make sure the same span id is used on the client and server, and that logic needs to kick in before the span starts in a naive impl.